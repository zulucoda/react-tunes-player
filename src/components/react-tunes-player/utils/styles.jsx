import styled from 'styled-components';
import device from './devices';
import {
  HIGH_VOLUME,
  LOW_VOLUME,
  MEDIUM_VOLUME,
  MUTE_VOLUME,
} from './constants';

export const ReactTunePlayerContainer = styled.div`
  position: fixed;
  bottom: -5px;
  visibility: hidden;
  width: 100%;
  -webkit-perspective: 900;
  perspective: 900;
  -webkit-perspective-origin: 80% 100%;
  perspective-origin: 80% 100%;
`;
export const TunesInner = styled.div`
  box-shadow: 0 0 6px 1px rgba(0, 0, 0, 0.25);
  height: 75px;
  visibility: visible;
  transition: -webkit-transform 200ms ease-out;
  transition: transform 200ms ease-out;
  -webkit-transform: translateY(100%);
  transform: translateY(100%);

  background-color: #f2f2f2;
  border-top: 1px solid #cecece;

  -webkit-transform: translateY(0);
  transform: translateY(0);
`;
export const TunesWrapper = styled.div`
  padding-bottom: 10px;
  visibility: visible;
  position: relative;
  height: 100%;
`;
export const MainWrapper = styled.div`
  display: flex;
  flex-direction: row;
  height: 100%;
  width: 100%;
`;
export const CurrentAlbumPlaying = styled.div`
  margin: 0 1rem 0 0;
  img {
    height: 75px;
    @media ${device.mobileS} {
      display: none;
    }

    @media ${device.mobileL} {
      display: inline;
    }
  }
`;
export const NowPlaying = styled.div`
  margin: 2rem 1rem 0 0;
  @media ${device.mobileS} {
    margin: 2rem 0 0 0;
    font-size: 0.7rem;
    flex-grow: 2;
  }

  @media ${device.mobileL} {
    display: inline;
    flex-grow: 1;
  }

  @media ${device.tablet} {
    display: inline;
    font-size: 1rem;
    margin: 2rem 1rem 0 0;
  }
`;
export const Controls = styled.div`
  margin: 1.2rem 0 0 0;

  @media ${device.mobileS} {
    margin: 1.8rem 0 0 0;
  }

  @media ${device.laptop} {
    margin: 1.2rem 0 0 0;
  }
`;
const Btn = styled.a`
  background-color: transparent;
  color: #231f20;
  font-size: 2em;
  padding: 0 1rem 0 1rem;

  @media ${device.mobileS} {
    font-size: 0.7rem;
    padding: 1rem 0.5rem 0 0.5rem;
  }

  @media ${device.tablet} {
    font-size: 1.3em;
    padding: 0 1rem 0 1rem;
  }

  @media ${device.laptop} {
    font-size: 2em;
  }

  > svg {
    width: 1rem;
    height: 1rem;
    cursor: pointer;
    fill: #594c4f;

    @media ${device.mobileS} {
      width: 0.7rem;
      height: 0.7rem;
    }

    @media ${device.tablet} {
      width: 0.8rem;
      height: 0.8rem;
    }

    @media ${device.laptop} {
      width: 1rem;
      height: 1rem;
    }

    &:hover {
      fill: #e93733;
    }
  }
`;
export const PreviousTune = styled(Btn)``;
export const NextTune = styled(Btn)``;
export const PlayTune = styled(Btn)`
  > svg {
    fill: ${props => (props.isPlaying ? '#E93733' : '#594C4F')};
  }
`;
export const PauseTune = styled(Btn)`
  > svg {
    fill: ${props => (!props.isPlaying ? '#E93733' : '#594C4F')};
  }
`;
export const SeekTune = styled.div`
  margin: 2rem 0 0 0;
  flex-grow: 2;

  @media ${device.mobileS} {
    display: none;
  }

  @media ${device.mobileL} {
    display: inline;
  }

  > progress {
    width: 100%;

    /* Turns off styling - not usually needed, but good to know. */
    appearance: none;
    -moz-appearance: none;
    -webkit-appearance: none;

    color: #e93733; /* IE 10 */

    /* Needs to be in here for Safari polyfill so background images work as expected. */
    background-size: auto;

    &::-moz-progress-bar {
      background-color: #e93733;
    }

    &::-webkit-progress-value {
      background-color: #e93733;
    }
  }
`;
export const SeekTime = styled.div`
  margin: 2rem 0 0 1rem;
  @media ${device.mobileS} {
    margin: 2rem 1rem 0 1rem;
    font-size: 0.7rem;
  }

  @media ${device.tablet} {
    font-size: 1rem;
  }
`;
export const Volume = styled(Controls)`
  @media ${device.mobileS} {
    display: none;
  }

  @media ${device.laptop} {
    display: inline;
  }
`;
export const HighVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === HIGH_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const MediumVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === MEDIUM_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const LowVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === LOW_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const MuteVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === MUTE_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const AudioWrapper = styled.div`
  margin: 0 auto;
  display: none;
`;
export const WarningWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;

  > svg {
    width: 3rem;
    height: 3rem;
    fill: #e93733;
  }
`;
export const WarningText = styled.span`
  padding: 0.5rem 0 0 0;
  font-size: 2rem;
  vertical-align: center;
`;
