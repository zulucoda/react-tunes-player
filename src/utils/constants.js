export const HIGH_VOLUME = 1;
export const MEDIUM_VOLUME = 0.5;
export const LOW_VOLUME = 0.3;
export const MUTE_VOLUME = 0;
