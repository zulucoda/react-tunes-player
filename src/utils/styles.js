import styled from 'styled-components';
import device from './devices';
import {
  HIGH_VOLUME,
  LOW_VOLUME,
  MEDIUM_VOLUME,
  MUTE_VOLUME,
} from './constants';

export const ReactTunePlayerContainer = styled.div`
  font-family: sans-serif;
  font-size: 1em;
  position: fixed;
  bottom: -5px;
  visibility: hidden;
  width: 100%;
  -webkit-perspective: 900;
  perspective: 900;
  -webkit-perspective-origin: 80% 100%;
  perspective-origin: 80% 100%;
`;
export const TunesInner = styled.div`
  box-shadow: 0 0 6px 1px rgba(0, 0, 0, 0.25);
  height: 75px;
  visibility: visible;
  transition: -webkit-transform 200ms ease-out;
  transition: transform 200ms ease-out;
  -webkit-transform: translateY(100%);
  transform: translateY(100%);

  background-color: #f2f2f2;
  border-top: 1px solid #cecece;

  -webkit-transform: translateY(0);
  transform: translateY(0);
`;
export const TunesWrapper = styled.div`
  padding-bottom: 10px;
  visibility: visible;
  position: relative;
  height: 100%;
`;
export const MainWrapper = styled.div`
  display: flex;
  flex-direction: row;
  height: 100%;
  width: 100%;
`;
export const CurrentAlbumPlaying = styled.div`
  margin: 0 1em 0 0;
  img {
    display: none;

    @media ${device.mobileL} {
      display: inline;
      height: 75px;
    }
  }
`;
export const NowPlaying = styled.div`
  margin: 2em 0 0 0;
  font-size: 0.8em;
  flex-grow: 1;
  flex-basis: 9.3em;

  @media ${device.mobileL} {
    flex-grow: 1;
  }

  @media ${device.tablet} {
    font-size: 1em;
    margin: 1.7em 1em 0 0;
  }

  @media ${device.laptop} {
    font-size: 1em;
    margin: 2em 1em 0 0;
  }
`;
export const Controls = styled.div`
  margin: 1.5em 0 0 0;
  flex-grow: 1;

  @media ${device.tablet} {
    margin: 1.7em 0 0 0;
  }
`;
const Btn = styled.a`
  background-color: transparent;
  color: #231f20;
  font-size: 1em;
  padding: 1em 0.5em 0 0.5em;

  @media ${device.tablet} {
    font-size: 1em;
    padding: 0 0.7em 0 0.7em;
  }

  @media ${device.laptop} {
    font-size: 1.5em;
    padding: 0 1em 0 1em;
  }

  > svg {
    cursor: pointer;
    fill: #594c4f;
    width: 0.7em;
    height: 0.7em;

    @media ${device.tablet} {
      width: 0.8em;
      height: 0.8em;
    }

    @media ${device.laptop} {
      width: 0.8em;
      height: 0.8em;
    }

    &:hover {
      fill: #e93733;
    }
  }
`;
export const PreviousTune = styled(Btn)``;
export const NextTune = styled(Btn)``;
export const PlayTune = styled(Btn)`
  > svg {
    fill: ${props => (props.isPlaying ? '#E93733' : '#594C4F')};
  }
`;
export const PauseTune = styled(Btn)`
  > svg {
    fill: ${props => (!props.isPlaying ? '#E93733' : '#594C4F')};
  }
`;
export const SeekTune = styled.div`
  display: none;
  @media ${device.tablet} {
    display: inline;
    margin: 1.7em 0 0 0;
    flex-grow: 2;
  }

  @media ${device.laptop} {
    display: inline;
    margin: 2em 0 0 0;
    flex-grow: 2;
  }
  > progress {
    width: 100%;

    /* Turns off styling - not usually needed, but good to know. */
    appearance: none;
    -moz-appearance: none;
    -webkit-appearance: none;

    color: #e93733; /* IE 10 */

    /* Needs to be in here for Safari polyfill so background images work as expected. */
    background-size: auto;

    &::-moz-progress-bar {
      background-color: #e93733;
    }

    &::-webkit-progress-value {
      background-color: #e93733;
    }
  }
`;
export const SeekTime = styled.div`
  margin: 2em 1em 0 1em;
  font-size: 0.8em;
  flex-basis: 5.5em;
  text-align: center;

  @media ${device.tablet} {
    font-size: 1em;
    margin: 1.7em 1em 0 1em;
  }

  @media ${device.laptop} {
    font-size: 1em;
    margin: 2em 1em 0 1em;
  }
`;
export const Volume = styled(Controls)`
  display: none;
  @media ${device.laptop} {
    display: inline;
  }
`;
export const HighVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === HIGH_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const MediumVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === MEDIUM_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const LowVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === LOW_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const MuteVolume = styled(Btn)`
  > svg {
    fill: ${props => (props.volume === MUTE_VOLUME ? '#E93733' : '#594C4F')};
  }
`;
export const AudioWrapper = styled.div`
  margin: 0 auto;
  display: none;
`;
export const WarningWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  > svg {
    width: 3em;
    height: 3em;
    fill: #e93733;
  }
`;
export const WarningText = styled.span`
  padding: 0.5em 0 0 0;
  font-size: 1em;
  vertical-align: center;
  p {
    padding: 0;
    margin: 0;
  }
`;
