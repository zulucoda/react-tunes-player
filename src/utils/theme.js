import styled from 'styled-components';
import device from './devices';

const DARK_COLOR = '#181818';
const LIGHT_COLOR = '#eef2f7';

export const PlayerGrid = styled.div`
  background: ${DARK_COLOR};
  display: grid;
  color: ${LIGHT_COLOR};
  grid-template-columns: 1fr 2fr 1fr;
  position: fixed;
  bottom: 0;
  width: 100%;
  margin: 0;
  padding: 0;
  height: 90px;
  grid-gap: 0.5em;

  // tablet
  @media ${device.tablet} {
    grid-gap: 0.5em;
    grid-template-columns: 2fr 1fr 3fr;
  }

  // desktop
  @media ${device.laptop} {
    grid-gap: 2em;
    grid-template-columns: 2fr 1fr 3fr 1fr;
  }
`;
