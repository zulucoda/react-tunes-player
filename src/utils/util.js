/**
 * Created by Muzikayise Flynn Buthelezi (zuluCoda) on 2019/08/18.
 * Copyright mfbproject.co.za - muzi@mfbproject.co.za
 * Copyright zulucoda - mfbproject
 */
const findIndex = (tunes, currentTune) =>
  tunes && tunes.length
    ? tunes.findIndex((t) => t.name === currentTune.name)
    : -1;

export const setPreviousTune = (tunes, currentTune, setCurrentTune) => {
  const previous = findIndex(tunes, currentTune) - 1;
  previous <= -1
    ? setCurrentTune(tunes[tunes.length - 1])
    : setCurrentTune(tunes[previous]);
};

export const setNextTune = (tunes, currentTune, setCurrentTune) => {
  const next = findIndex(tunes, currentTune) + 1;
  next >= tunes.length ? setCurrentTune(tunes[0]) : setCurrentTune(tunes[next]);
};

export const formatSeekTime = (timeToFormat) =>
  timeToFormat > 0 ? (timeToFormat / 60).toFixed(2) : 0;
