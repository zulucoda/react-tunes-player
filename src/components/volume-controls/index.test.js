import React from 'react';
import VolumeControls from '.';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LOW_VOLUME } from '../../utils/constants';

describe('Components - Volume Controls - Render Test', () => {
  let volumeMuteMock;
  let volume0Mock;
  let volume1Mock;
  let volume2Mock;
  beforeEach(() => {
    volumeMuteMock = jest.fn();
    volume0Mock = jest.fn();
    volume1Mock = jest.fn();
    volume2Mock = jest.fn();
  });

  test('should render player controls', () => {
    const { container } = render(
      <VolumeControls
        volumeMute={volumeMuteMock}
        volumeMedium={volume1Mock}
        volumeHigh={volume2Mock}
        volumeLow={volume0Mock}
        volume={LOW_VOLUME}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
