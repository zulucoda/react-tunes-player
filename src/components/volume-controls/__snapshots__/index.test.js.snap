// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Volume Controls - Render Test should render player controls 1`] = `
<div>
  <div
    class="sc-bczRLJ cKJGjv volume-controls"
  >
    <ul
      class="sc-gsnTZi jMoirn volume-controls-list"
    >
      <li
        class="sc-dkzDqf sc-gKXOVf RRkiZ cMCDLz volume-controls-list-x"
      >
        <svg
          class="volume-x"
          fill="none"
          height="24"
          role="volume-x"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="11 5 6 9 2 9 2 15 6 15 11 19 11 5"
          />
          <line
            x1="22"
            x2="16"
            y1="9"
            y2="15"
          />
          <line
            x1="16"
            x2="22"
            y1="9"
            y2="15"
          />
        </svg>
      </li>
      <li
        class="sc-dkzDqf sc-jSMfEi RRkiZ jIdZZO volume-controls-list-0"
      >
        <svg
          class="volume"
          fill="none"
          height="24"
          role="volume"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="11 5 6 9 2 9 2 15 6 15 11 19 11 5"
          />
        </svg>
      </li>
      <li
        class="sc-dkzDqf sc-eCYdqJ RRkiZ dgKCTI volume-controls-list-1"
      >
        <svg
          class="volume-1"
          fill="none"
          height="24"
          role="volume-1"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="11 5 6 9 2 9 2 15 6 15 11 19 11 5"
          />
          <path
            d="M15.54 8.46a5 5 0 0 1 0 7.07"
          />
        </svg>
      </li>
      <li
        class="sc-dkzDqf sc-hKMtZM RRkiZ kTNKeB volume-controls-list-2"
      >
        <svg
          class="volume-2"
          fill="none"
          height="24"
          role="volume-2"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="11 5 6 9 2 9 2 15 6 15 11 19 11 5"
          />
          <path
            d="M15.54 8.46a5 5 0 0 1 0 7.07"
          />
          <path
            d="M19.07 4.93a10 10 0 0 1 0 14.14"
          />
        </svg>
      </li>
    </ul>
  </div>
</div>
`;
